common/beerocks/tlvf/src/include/beerocks/tlvf/beerocks_message.h

```diff
@@ -139,7 +175,15 @@ class message_com {
             return nullptr;
         }
 
-        return add_intel_vs_data<T>(cmdu_tx, tlvhdr, id);
+        auto beerocks_header = add_intel_vs_data<T>(tlvhdr, id);
+        cmdu_tx.header = beerocks_header;
+
+        // According to C++'03 Standard 14.2/4:
+        // When the name of a member template specialization appears after . or -> in a postfix-expression,
+        // or after nested-name-specifier in a qualified-id, and the postfix-expression or qualified-id
+        // explicitly depends on a template-parameter (14.6.2), the member template name must be prefixed by
+        // the keyword template. Otherwise the name is assumed to name a non-template.
+        return beerocks_header->template getClass<T>();
```

Comment:
So if we remove the non-templated getClass, this wouldn't be needed anymore, right?

If we don't remove it, we could still rename it.

Reply:
Not sure I understand your point, let's discuss it later in the final PR.